Add first REST Point API for calculating prayer times

lucernae opened the pull request:
# Changes

- Use composer library: geniusts/prayer-times@1.0.4
- Create PrayerTimesController
- Create routing files to `/api/praytimes/`
- Define new Calculation method to hold values for Kemenag methods
- Add Makefile to easily run webserver for local development

# Notes

- [ ] Value buat ngeset sudut shubuh ama isha dari kemenag dapetnya gimana? Dah nyari-nyari ga nemu.
- [ ] Librarynya ga selengkap praytimes.org yang asli. contohnya, ga bisa tahu midnight, etc.

# Ref

- Librarynya: https://github.com/GeniusTS/prayer-times
- Sumber algoritma: https://github.com/QuantumPrayerTimes/prayertimes
- Mending jadiin meta ticketnya buat referensi sumber. Soalnya algoritmanya dah gaib pertanggunganjawabnya.

lucernae committed:
Add first REST Point API for calculating prayer times

- Use composer library: geniusts/prayer-times@1.0.4
- Create PrayerTimesController
- Create routing files to `/api/praytimes/`
- Define new Calculation method to hold values for Kemenag methods
- Add Makefile to easily run webserver for local development

lucernae commented on app/Helpers/PrayerTimes/Methods/Kemenag.php:
```diff
@@ -0,0 +1,60 @@
+     * @var array<float>
+     */
+    protected $angles = [
+        'fajr' => 18.0,
```
Ini valuenya ntar dapet darimana, gan?
Sekarang default dulu.

aragen replied:
value apa maksudnya? anglesnya masing2 pake konvensi. Kalo kemenag: Fajr 20 degree, Isha 18 degree

lucernae replied:
Iya, maksdnya value ini. Yawda, diganti dulu

lucernae commented on app/Http/Controllers/PrayerTimesController.php:
```diff
@@ -0,0 +1,167 @@
+        $date = $args["date"];
+        $timezone = $args["timezone"];
+        $time_format = $args["time_format"];
+        $method = self::get_if_exists($args, "method");
```
Ada cara yang lebih baik ga sih buat ngambil value tapi ga error kalo ga ada? :v

aragen replied:
please add default value referring to #2

lucernae commented on app/Http/Controllers/PrayerTimesController.php:
```diff
@@ -0,0 +1,167 @@
+
+        /**
+         * Response format:
+         * {
```
Baru draft. Maneh maunya gimana ntar?

aragen replied:
refer ke #2 ya.
urang bingung kalo mau urg kasih notes panjang, misal, calculation method itu ada apa aja dan valuenya apa aja. urg taro di mana ya?

lucernae commented on bootstrap/app.php:
```diff
@@ -97,6 +97,7 @@
     'namespace' => 'App\Http\Controllers',
 ], function ($router) {
     require __DIR__.'/../routes/web.php';
+    require __DIR__.'/../routes/API/praytimes.php';
```
Nge-addnya harus kayak gini, gitu? Jelek amat...

lucernae reviewed (commented):
Mangga direvyuh.. @aragen

lucernae commented:
Kalo mau ngecek, contoh request:

```
target: /api/praytimes/calculate
request-body:

{"lat":-6.914744,"lng": 107.609810,"date": "2018-12-23","method":"kemenag","timezone":7,"time_format":"H:i"}

```

urang pake postman doang.

lucernae committed:
Update API values

- Change Kemenag default angle values
- Configure default settings
- Handle params for GET query strings, but overrideable from request body

lucernae commented:
Applied #2 and #1 

Sama, sekarang paramnya bisa ditaro di query string.
